### PR TITLE
Platform/gamecore

### PR DESCRIPTION
**Pull request by Adrian1066**
# **Please read the [Contributing guide](CONTRIBUTING.md) before making a PR.**

* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
This PR adds support for the XboxSeries platform to the SRP packages and the PPv2 package.

---
### Testing status
Manual testing with HDRP packages
Testing of HDRP, URP, ShaderGraph and VFXGraph run locally.
Tested on Xbox to check for regressions

---
### Comments to reviewers
PPV2 changes

The Xbox build target is wrapped to allow this package to work even in older versions of Unity as PS5 will not show up as a valid enum member there.

### CODE REVIEW

File: com.unity.render-pipelines.core/Runtime/Textures/TextureXR.cs
Comment (robinb-u3d) [inline com.unity.render-pipelines.core/Runtime/Textures/TextureXR.cs:43]: The return true shouldn't be in the middle here as that is what the whole list here is for.
So now it says
case MSPLATFORMS
    return true;
case OTHERVRPLATFORMS
    return true;
default:
     return false

**Review by theo-pnv — APPROVED** [review]
Approving on behalf of the gfx-sdets team **only for this file** https://github.com/Unity-Technologies/Graphics/pull/3292/files#diff-75b9357cf92d6fe2352865e0ae25478fb940ae0ac8509679db27f00efff68011 (Since I believe that's what auto added us to the reviewers). 
Please wait for other relevant approvals before merging.

File: com.unity.render-pipelines.high-definition/Documentation~/Building-For-Consoles.md
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Documentation~/Building-For-Consoles.md:3]: Is Scarlett a user facing name? Shouldn't this be Xbox One Series X/S ?

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.hlsl
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.hlsl:13]: :-( So we did not find the root issue yet?

File: com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/UpsampleTransparent.shader
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/UpsampleTransparent.shader:43]: This is not needed anymore, I fixed this in master to have BILINEAR be the default when nothing is defined.

File: com.unity.visualeffectgraph/Shaders/ParticleLinesSW/PassForward.template
Comment (FrancescoC-unity) [inline com.unity.visualeffectgraph/Shaders/ParticleLinesSW/PassForward.template:16]: Why here we have gamecore and not xbox? I did not look at the reference case, but I wonder if it was PSSL specific only?

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.hlsl
Comment (Adrian1066) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.hlsl:13]: Unfortunately not yet.

File: com.unity.visualeffectgraph/Shaders/ParticleLinesSW/PassForward.template
Comment (Adrian1066) [reply com.unity.visualeffectgraph/Shaders/ParticleLinesSW/PassForward.template:16]: Doesn't seem to be an issue on Xbox, but GPU hangs on GameCore

File: com.unity.render-pipelines.high-definition/Documentation~/Building-For-Consoles.md
Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Documentation~/Building-For-Consoles.md:3]: @JordanL8

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/scrbound.compute
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/scrbound.compute:122]: This comment/define shouldn't be here but in the GameCore specific include

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/VolumetricLighting.compute
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/VolumetricLighting.compute:88]: cc @EvgeniiG for the Zbinning PR

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/PathTracing/Shaders/PathTracingBSDF.hlsl
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/PathTracing/Shaders/PathTracingBSDF.hlsl:113]: @eturquin can you report those warning fix in a separate PR for path tracer? thanks

File: com.unity.render-pipelines.universal/Documentation~/Building-For-Consoles.md
Comment (sebastienlagarde) [inline com.unity.render-pipelines.universal/Documentation~/Building-For-Consoles.md:8]: Same question about documentation than HDPR

File: com.unity.render-pipelines.high-definition/Documentation~/Building-For-Consoles.md
Comment (JordanL8) [reply com.unity.render-pipelines.high-definition/Documentation~/Building-For-Consoles.md:3]: Yes we should use the user-facing names here. So Xbox Series X/S instead of scarlett

File: com.unity.template-hd/Packages/com.unity.template.hd/Documentation~/hd.md
Comment (JordanL8) [inline com.unity.template-hd/Packages/com.unity.template.hd/Documentation~/hd.md:17]: Should this be **Xbox Series X, Xbox Series S,** instead of **Xbox Series,**?

**Review by sebastienlagarde — CHANGES_REQUESTED** [review]
The PR miss modification of shader graph for HDPR. There is a xboxseries  to generate with the shader. 

See Graphics\com.unity.shadergraph\Editor\Generation\Enumerations\Platform.cs

**Review by sebastienlagarde — COMMENTED** [review]
ok, it appear that the change was already done in platform.cs I was just misreading.

File: com.unity.render-pipelines.high-definition/Documentation~/Building-For-Consoles.md
Comment (CharlieBeatnik) [reply com.unity.render-pipelines.high-definition/Documentation~/Building-For-Consoles.md:3]: So currently the Unity Game Core editor extensions we provide to our users are called:

- Game Core - Scarlett
- Game Core - Xbox One

All of our documentation and C# defines also refer to the new Xbox as "Scarlett". So for consistency I'd like HDRP to keep using the Scarlett name in its documentation.

In the future we may move everything over to using "Xbox Series", but we'll do an en masse rename if this happens.

**Review by sebastienlagarde — APPROVED** [review]

**Review by FrancescoC-unity — APPROVED** [review]

**Review by robinb-u3d — APPROVED** [review]

**Review by phi-lira — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/PathTracing/Shaders/PathTracingBSDF.hlsl
Comment (eturquin) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/PathTracing/Shaders/PathTracingBSDF.hlsl:113]: Will do

File: com.unity.render-pipelines.universal/Documentation~/Building-For-Consoles.md
Comment (Adrian1066) [reply com.unity.render-pipelines.universal/Documentation~/Building-For-Consoles.md]: Scarlett names have been replaced with Xbox Series

File: com.unity.template-hd/Packages/com.unity.template.hd/Documentation~/hd.md
Comment (Adrian1066) [reply com.unity.template-hd/Packages/com.unity.template.hd/Documentation~/hd.md:17]: Xbox Series covers both machines, as they are a single platform.

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/HDShadow.hlsl
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/HDShadow.hlsl:97]: isn't this working on Xbox? was guessing only gamecore is a problem right?

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/HDShadow.hlsl:97]: This is an old problem with just the shadow data scalarization, FXC fails hard with the eye shader here.

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/HDShadow.hlsl:97]: If it is a fix for xbox, then we should do a separate PR, it shouldn't be part of the GameCore PR. But yes, as we backport up to 2020.2 it is ok, eye shader wasn't present before

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/HDShadow.hlsl:97]: we should add a comment for the shader data scalarezitation for eye sahder, so we don't forget it is because of this.

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/HDShadow.hlsl:97]: It was there before and there is a comment up above :)

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/HDShadow.hlsl:97]:
```
    // We need to disable the scalarization here on xbox due to bad code generated by FXC for the eye shader.
    // This shouldn't have an enormous impact since with Area lights we are already exploded in VGPR by this point.

```

**Review by sebastienlagarde — APPROVED** [review]

**Review by esmelusina — APPROVED** [review]
Looks fine for SG.

File: com.unity.visualeffectgraph/Shaders/ParticleLinesSW/PassForward.template
Comment (PaulDemeulenaere) [reply com.unity.visualeffectgraph/Shaders/ParticleLinesSW/PassForward.template:16]: +1 This is probably a common architecture issue, we are still investigating.

**Pull request merged**
